lib/RoleDetails/RoleDetails.js

```diff
@@ -190,8 +189,9 @@ export const RoleDetails = ({
     } catch (error) {
       handleError(MUTATION_ACTION_TYPE.delete, error);
     }
-  }, [deleteRole, deleteSharedRole, handleError, isRoleShared, role, roleId, showCallout]);
+  }, [deleteRole, deleteSharedRole, handleError, isRoleShared, onClose, role, roleId, showCallout]);
```

Comment:
I would suggest for better readability 
```suggestion
  }, [
    deleteRole,
    deleteSharedRole,
    handleError,
    isRoleShared,
    onClose,
    role,
    roleId,
    showCallout,
  ]);
```